Feature/sawtooth with api

ivanminutillo opened the pull request:
First implementation of needed endpoints to connect swapi with the petition transactor processor.
New endpoints are:
- ```Create petition```
- ```Sign petition```
- ```Tally petition```
- ```Count petition```

Under the original `Transactions` endpoint is possible also to `list petitions` and `get a petition` by petition_id.

### WIP :
- improving schema and errors management based on the Petition TP logic.
- In the coming period we may want to isolate different implementation of swapi based on their contexts.
Eg. `Petition` and `Transactions` can become 2 separate implementation, that can be used together if needed, `ValueFlows`can sooner be the next.
As we keep developing swapi and envision to use it as blockchain interop API layer and schema checker, we think it will be foundamental to start experimenting with a more modularity architecture.

aspasia committed:
Import cleanup and split handler into two parts (core created)

aspasia committed:
FIX: project file with splitting handler

aspasia committed:
WIP: setting up endpoint for Sawtooth

aspasia committed:
Linter cleanup/suggestions

aspasia committed:
FIX: simple list transactions now works

aspasia committed:
FIX: Empty vector was causing swagger to crash

aspasia committed:
Fixed travis config due to travis CONSTANT changes

aspasia committed:
Port to work with zen tooth tp

aspasia committed:
tiny linter updates

aspasia committed:
Now the transaction get works for sawtooth

aspasia committed:
FIX: tests to work with the new schemas

aspasia committed:
FIX: small fixes after merging

aspasia committed:
Sawtooth create petition schemas added

aspasia committed:
WIP: started with adding the endpoint for create petition. Some corrections needed for it to work

aspasia committed:
FIX: reading petition api credentials and creating connection

aspasia committed:
WIP: fixing the CreatePetition schema

aspasia committed:
WIP: creating the petition json from a stored template.

aspasia committed:
FIX: create petition now works wit schema checks

Commit:
added sign petition API

Commit:
import the sign petition json

Commit:
added schema for sign petition

Commit:
construct tally petition json

Commit:
added schema checks for Tally petition json and CountPetitionResponse

Commit:
Added endpoints for tally and count

puria commented:
@ivanminutillo ❤️

aspasia commented on src/social_wallet_api/handler.clj:
```diff
@@ -475,6 +475,31 @@ Returns the DB entries that were created.
+    :return CreatePetitionResponse
+    :body [query NewPetition]
+    :summary "Sign an existing petition"
+    :description "TODO: "
```
Please add description with parameter options before merging

aspasia commented on src/social_wallet_api/handler.clj:
```diff
@@ -469,37 +469,73 @@ Returns the DB entries that were created.
+       :body [query NewPetition]
+       :summary "Tally an existing petition"
+       :description "TODO: "
+       (with-error-responses swapi/connections query
```
Same here

aspasia commented on src/social_wallet_api/handler.clj:
```diff
@@ -469,37 +469,73 @@ Returns the DB entries that were created.
+       :body [query NewPetition]
+       :summary "Count a tallied petition"
+       :description "TODO: "
+       (with-error-responses swapi/connections query
```
And here

aspasia reviewed (commented):
@ivanminutillo all looks good apart from the fact that we need the descriptions to make sure the SWAPI is self-documented. Also wondering if we should push the templates somewhere. I will now test locally, but I am not sure if we should merge or just keep it on a separate branch for now. The SWAPI is created generic and the petitions are very specific. Maybe we should wait until after we create the SWAPI families. Lets discuss

Commit:
Added description for each new Petition endpoint, and added list and get endpoints as well under petition

Commit:
Updated SawtoothPaging schema and created custom PetitionQuery Schema

Commit:
updated swapi tests to work with latest schema
